### PR TITLE
30 course card

### LINKED ISSUE #30: Course Card
Use MUI for easy card styling. 

Some complexity with hover; should be a common use case for expanding a card (probably tutorials online) 
- Consider attaching an event handler on the component for hover which runs a function to display additional data 
- Consider conditionally rendering two versions of the components (closed and expanded) because on event handler onHover  

Once done, we can create a new branch to add this component inside the .map that renders from the search bar component.

### PR DESCRIPTION
**Pull request by Inioluwa21**
Things might not be perfect yet as I would need to integrate the search bar to see how it looks all together.

image

image

### CODE REVIEW

File: client/src/components/courseCard.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      onMouseLeave={handleMouseLeave}
+      className="course-card"
+    >
+      <CardContent className="course-card-content">
+        <div className="course-title">
+          <p>Course name {props.name}</p>
+        </div>
+        <div noWrap={!isHovered} className="course-body">
+          <p>
+            Principles and techniques for developing software combined with the
```
Comment (marmikpatel19) [inline client/src/components/courseCard.jsx]: I assume this is an example. If so, we shouldn't commit it

File: client/src/components/coursePage.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React, { useState } from 'react';
+import CourseCard from './courseCard';
+import '../stylings/coursePage.css';
+
+const courses = [
```
Comment (marmikpatel19) [inline client/src/components/coursePage.jsx]: Similar to above comments. Keep temporary changes like temp data local and not commit them.

**Review by marmikpatel19 — CHANGES_REQUESTED** [review]

**Comment by marmikpatel19** [comment]
Search is not in main; let's merge that into this branch to see if things look find with that and footer all together. This may cause (necessary) styling changes.

**Comment by marmikpatel19** [comment]
The font in the pictures looks off. Can you confirm that it's the font being used app-wide?

File: client/src/components/courseCard.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            practical experience of creating a mid-size software system as a
+            member of a software development team. Includes: teamwork; projects,
+            planning and process; users and requirements; use cases; modeling;
+            quality; software architecture; testing; GUI design, design
+            principles, patterns and implementation; ethics; professionalism.
+            {props.body}
```
Comment (marmikpatel19) [inline client/src/components/courseCard.jsx]: Course component seems to missing all the other fields like hyperlink, prerequisites, dependencies, etc. We (I) will do the actual connect the endpoint in another issue, but the course card needs to be ready to display all the necessary content.

**Comment by marmikpatel19** [comment]
Colors look quite dull. After adding in all the additional fields, try adding in bits of our primary and/or secondary colors as defined in App.css

**Comment by marmikpatel19** [comment]
When done, please attach responsive photos with all new data mocked, as you've done above. Consider cases where certain data is missing, such as prerequisites. In the case of missing data, something else should be displayed to indicate so.

**Comment by Inioluwa21** [comment]
image

**Comment by Inioluwa21** [comment]
image

**Review by marmikpatel19 — COMMENTED** [review]
The coursePage.jsx component isn't actually a page--it's embedded inside of the home.jsx page. Could we change its name it be something like courses.jsx and the courseCard.jsx to course.jsx?

**Review by marmikpatel19 — APPROVED** [review]

**Comment by marmikpatel19** [comment]
Beyond the comment I left above, everything else looks good--great work! This was definitely one of the bigger pieces.

**Pull request merged**